features/networking/ovn-egress-ip.feature

```diff
@@ -482,3 +482,151 @@ Feature: OVN Egress IP related features
+
+    #Create egress ip object
+    When I obtain test data file "networking/ovn-egressip/egressip1.yaml"
+    And I replace lines in "egressip1.yaml":
+      | 172.31.249.227 | "<%= cb.valid_ip %>" |
+    And I run the :create admin command with:
+      | f | egressip1.yaml |
+    And admin ensures "egressip" egress_ip is deleted after scenario
+
+    #Label EgressIP nodes.
+    Given I store the schedulable nodes in the :nodes clipboard
+    Then label "k8s.ovn.org/egress-assignable=true" is added to the "<%= cb.nodes[0].name %>" node
+
+    # Check source ip is egress ip
+    Then I wait up to 20 seconds for the steps to pass:
+    """
+    When I execute on the pod:
+      | curl | -s | --connect-timeout | 5 | <%= cb.ipecho_url %> |
+    Then the step should succeed
+    And the output should contain "<%= cb.valid_ip %>"
+    """
+
+    #Remove egress ip object and recreate it
+    Given admin ensures "egressip" egress_ip is deleted
+    When I obtain test data file "networking/ovn-egressip/egressip1.yaml"
+    And I replace lines in "egressip1.yaml":
+      | 172.31.249.227 | "<%= cb.valid_ip %>" |
+    And I run the :create admin command with:
+      | f | egressip1.yaml |
+    Then the step should succeed
+
+    # Check source ip is egress ip
+    Then I wait up to 20 seconds for the steps to pass:
+    """
+    When I execute on the pod:
+      | curl | -s | --connect-timeout | 5 | <%= cb.ipecho_url %> |
+    Then the step should succeed
+    And the output should contain "<%= cb.valid_ip %>"
+    """
+
+  # @author huirwang@redhat.com
+  # @case_id OCP-33710
+  @admin
+  @destructive
+  Scenario: An EgressIP object can not have multiple egress IP assignments on the same node
+    Given I store the schedulable workers in the :nodes clipboard
+    Then label "k8s.ovn.org/egress-assignable=true" is added to the "<%= cb.nodes[0].name %>" node
+
+    #Get unused IP as egress ip
+    Given I store a random unused IP address from the reserved range to the clipboard
+
+    #Create first egress ip object with two EgressIPs
+    When I obtain test data file "networking/ovn-egressip/egressip2.yaml"
+    And I replace lines in "egressip2.yaml":
+      | 172.31.249.227 | "<%= cb.valid_ips[0] %>" |
+      | 172.31.249.228 | "<%= cb.valid_ips[1] %>" |
+    And I run the :create admin command with:
+      | f | egressip2.yaml |
+    And admin ensures "egressip2" egress_ip is deleted after scenario
+
+    # Check only one IP assigned.
+    When I run the :get admin command with:
+      | resource       | egressip                    |
+      | resource_name  | egressip2                   |
+      | o              | jsonpath={.status.items[*]} |
+    Then the step should succeed
+    And evaluation of `@result[:response].chomp.match(/\d{1,3}\.\d{1,3}.\d{1,3}.\d{1,3}/).length` is stored in the :egressip_num clipboard
+    Then the expression should be true> cb.egressip_num == 1
+
+  # @author huirwang@redhat.com
+  # @case_id OCP-33617
+  @admin
```

Comment:
This scenario says "common user" but tagged as `@admin` , is that correct ? Or we need to tag it as `@user`? Curious. :thinking:

Reply:
@kedark3 if you look at the step_definiton for `Given I select a random node's host` it requires `@admin`

Reply:
oh ok @pruan-rht . But then how does it ensure ```#Label nodes with normal user
    When I run the :label client command with:``` 
the mention `normal user` is only in the comment, and not seeing actual code that invokes it as normal user. Wonder what am I missing.

Reply:
@kedark3  "label client " means it will use the common user,  not admin. See the logs result:
 oc label node huirwang-test-4b7c6-master-0 k8s.ovn.org/egress-assignable\=true --kubeconfig=/home/jenkins/ws/workspace/Runner-v3-smoke/workdir/ocp4_testuser-2.kubeconfig
      
      STDERR:
      Error from server (Forbidden): nodes "huirwang-test-4b7c6-master-0" is forbidden: User "testuser-2" cannot get resource "nodes" in API group "" at the cluster scope

Reply:
That makes sense now. Thank you.